src/writer.rs

```diff
@@ -255,6 +330,9 @@ impl FdtWriter {
     /// `name` - name of the node; must not contain any NUL bytes.
     pub fn begin_node(&mut self, name: &str) -> Result<FdtWriterNode> {
         let name_cstr = CString::new(name).map_err(|_| Error::InvalidString)?;
+        if !node_name_valid(name) {
```

Comment:
Can we add a comment here that in case the user uses unit-address, then they must provide a valid one according to the specification (and then add a link to the spec)?

Reply:
Sure, added a comment to this effect.